graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,126 @@
+        """Return a human-readable representation of the Statistics object."""
+        return self.__str__()
+
+    @abstractmethod
+    def get_class_count(self, class_name):
+        """Return how many vertex or edge instances have, or inherit, the given class name.
+
+        Args:
+            class_name: str, either a vertex class name or an edge class name defined in the
+                        GraphQL schema.
+
+        Returns:
+            int, count of vertex or edge instances having, or inheriting, the given class name.
+
+        Raises:
+            AssertionError, if the count statistic for the given vertex/edge class does not exist.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_vertex_edge_vertex_count(
+        self, vertex_source_class_name, edge_class_name, vertex_target_class_name
+    ):
+        """Return count of edge_class edges connecting vertex_source and vertex_target vertices.
+
+        This statistic is optional, as the estimator can roughly predict this statistic using
+        get_class_count(). In some cases of traversal between two vertices using an edge connecting
+        the vertices' superclasses, the estimates generated using get_class_count() may be off by
+        several orders of magnitude. In such cases, this statistic should be provided.
+
+        Args:
+            vertex_source_class_name: str, vertex class name defined in the GraphQL schema.
+            edge_class_name: str, edge class name defined in the GraphQL schema.
+            vertex_target_class_name: str, vertex class name defined in the GraphQL schema.
+
+        Returns:
+            - int, count of edges of class edge_class with the two vertex classes as its endpoints
+                   if the statistic exists.
+            - None otherwise.
+        """
+        return None
+
+    @abstractmethod
+    def get_distinct_field_values_count(self, vertex_name, field_name):
+        """Return the count of distinct values a vertex's property field has over all instances.
+
+        This statistic helps estimate the result size of @filter directives used to filter
+        values using equality operators like '=', '!=', and 'in_collection'.
+
+        Args:
+            vertex_name: str, name of a vertex defined in the GraphQL schema.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - int, count of distinct values of that vertex's property field if the statistic exists.
+            - None otherwise.
+        """
+        return None
+
+
+class LocalStatistics(Statistics):
+    """Statistics class that receives all statistics at initialization, storing them in-memory."""
+    def __init__(
+        self, class_counts, vertex_edge_vertex_counts=None,
+        distinct_field_values_counts=None
+    ):
+        """Initialize statistics with the given data.
+
+        Args:
+            class_counts: dict, str -> int, mapping vertex/edge class name to class count.
```

Comment:
Do we explicitly define anywhere what class count is? I hadn't looked at this code in a while and thought for a second class count was the number of classes in the schema, but then realized it actually means number of vertices or edges of that class.

Reply:
Agreed. Class_counts was a bit of a holdover term that cost estimation used before I even started working on this, so I've been a bit reluctant to change it.
I've changed this line to 
```
            class_counts: dict, str -> int, mapping vertex/edge class name to count of
                instances of that class.
```
Thanks for your feedback!